core/templates/dev/head/pages/exploration-editor-page/services/exploration-rights.service.spec.ts

```diff
@@ -27,73 +27,275 @@ import { UpgradedServices } from 'services/UpgradedServices';
-      expect(ers.isCloned()).toBe(true);
-      expect(ers.clonedFrom()).toEqual('1234');
+    clearWarningsSpy = spyOn(als, 'clearWarnings').and.callThrough();
+  }));
+
+  it('should correctly initializes the service', function() {
+    expect(ers.ownerNames).toBeUndefined();
+    expect(ers.editorNames).toBeUndefined();
+    expect(ers.voiceArtistNames).toBeUndefined();
+    expect(ers.viewerNames).toBeUndefined();
+    expect(ers.isPrivate()).toBe(false);
+    expect(ers.isPublic()).toBe(false);
+    expect(ers.clonedFrom()).toBeUndefined();
+    expect(ers.isCommunityOwned()).toBeUndefined();
+    expect(ers.viewableIfPrivate()).toBeUndefined();
+
+    ers.init(
+      sampleDataResults.rights.owner_names,
+      sampleDataResults.rights.editor_names,
+      sampleDataResults.rights.voice_artist_names,
+      sampleDataResults.rights.viewer_names,
+      sampleDataResults.rights.status,
+      sampleDataResults.rights.cloned_from,
+      sampleDataResults.rights.community_owned,
+      sampleDataResults.rights.viewable_if_private
+    );
+
+    expect(ers.ownerNames).toEqual(sampleDataResults.rights.owner_names);
+    expect(ers.editorNames).toEqual(sampleDataResults.rights.editor_names);
+    expect(ers.voiceArtistNames).toEqual(
+      sampleDataResults.rights.voice_artist_names);
+    expect(ers.viewerNames).toEqual(sampleDataResults.rights.viewer_names);
+    expect(ers.isPrivate()).toEqual(true);
+    expect(ers.clonedFrom()).toEqual(sampleDataResults.rights.cloned_from);
+    expect(ers.isCommunityOwned()).toBe(
+      sampleDataResults.rights.community_owned);
+    expect(ers.viewableIfPrivate()).toBe(
+      sampleDataResults.rights.viewable_if_private);
+  });
+
+  it('should reports the correct cloning status', function() {
+    ers.init(['abc'], [], [], [], 'public', '1234', true);
+    expect(ers.isCloned()).toBe(true);
+    expect(ers.clonedFrom()).toEqual('1234');
+
+    ers.init(['abc'], [], [], [], 'public', null, true);
+    expect(ers.isCloned()).toBe(false);
+    expect(ers.clonedFrom()).toBeNull();
+  });
+
+  it('should reports the correct community-owned status', function() {
+    ers.init(['abc'], [], [], [], 'public', '1234', false);
+    expect(ers.isCommunityOwned()).toBe(false);
+
+    ers.init(['abc'], [], [], [], 'public', '1234', true);
+    expect(ers.isCommunityOwned()).toBe(true);
+  });
 
-      ers.init(['abc'], [], [], [], 'public', null, true);
-      expect(ers.isCloned()).toBe(false);
-      expect(ers.clonedFrom()).toBeNull();
+  it('should reports the correct derived statuses', function() {
+    ers.init(['abc'], [], [], [], 'private', 'e1234', true);
+    expect(ers.isPrivate()).toBe(true);
+    expect(ers.isPublic()).toBe(false);
+
+    ers.init(['abc'], [], [], [], 'public', 'e1234', true);
+    expect(ers.isPrivate()).toBe(false);
+    expect(ers.isPublic()).toBe(true);
+  });
+
+  it('should reports correcty if exploration rights is viewable when' +
+    ' private', function() {
```

Comment:
I think the ' should align with the line above?

Reply:
Done.